EventFilter/HcalRawToDigi/plugins/alpaka/HcalDigisProducerPortable.cc

```diff
@@ -0,0 +1,200 @@
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiHostCollection.h"
+#include "DataFormats/HcalDigi/interface/alpaka/HcalDigiDeviceCollection.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class HcalDigisProducerPortable : public stream::EDProducer<> {
+  public:
+    explicit HcalDigisProducerPortable(edm::ParameterSet const& ps);
+    ~HcalDigisProducerPortable() override = default;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+  private:
+    void produce(device::Event&, device::EventSetup const&) override;
+
+  private:
+    // input product tokens
+    edm::EDGetTokenT<HBHEDigiCollection> hbheDigiToken_;
+    edm::EDGetTokenT<QIE11DigiCollection> qie11DigiToken_;
+
+    // type aliases
+    using HostCollectionPhase1 = hcal::Phase1DigiHostCollection;
+    using HostCollectionPhase0 = hcal::Phase0DigiHostCollection;
+
+    using DeviceCollectionPhase1 = hcal::Phase1DigiDeviceCollection;
+    using DeviceCollectionPhase0 = hcal::Phase0DigiDeviceCollection;
+
+    // output product tokens
+    device::EDPutToken<DeviceCollectionPhase1> digisF01HEToken_;
+    device::EDPutToken<DeviceCollectionPhase0> digisF5HBToken_;
+    device::EDPutToken<DeviceCollectionPhase1> digisF3HBToken_;
+
+    struct ConfigParameters {
+      uint32_t maxChannelsF01HE, maxChannelsF5HB, maxChannelsF3HB;
+    };
+    ConfigParameters config_;
+  };
+
+  void HcalDigisProducerPortable::fillDescriptions(edm::ConfigurationDescriptions& confDesc) {
+    edm::ParameterSetDescription desc;
+
+    // FIXME
```

Comment:
Is there still something that needs to be fixed?

Reply:
removed